File: biddo_web/src/core/repositories/filter.ts
import { FilterItem } from '../domain/filter'
import { RequestMaker, RequestType } from '../services/request-maker'

class FiltersRepository {
  private basePath = '/filters'

  public async create(item: FilterItem): Promise<FilterItem | null> {
    try {
      const result = (await RequestMaker.makeRequest({
        path: this.basePath,
        method: RequestType.POST,
        payload: JSON.stringify(item),
        contentType: 'application/json',
      })) as { filterItem: Record<string, unknown> }
      return FilterItem.fromJSON(result.filterItem)
    } catch (error) {
      console.error('Error creating filter:', error)
      return null
    }
  }

  public async delete(filterId: string): Promise<boolean> {
    try {
      await RequestMaker.makeRequest({
        path: `${this.basePath}/${filterId}`,
        method: RequestType.DELETE,
      })
      return true
    } catch (error) {
      console.error('Error deleting filter:', error)
      return false
    }
  }
}

const FiltersRepositoryInstance = new FiltersRepository()
export { FiltersRepositoryInstance as FiltersRepository }

File: biddo_client/lib/core/repositories/filter.dart
import 'package:get/get.dart';

import '../models/filter.dart';
import 'base.dart';

class FiltersRepository {
  var dio = Get.find<Api>();

  Future<bool> create(FilterItem item) async {
    try {
      await dio.api.post('/filters', data: item.toJSON());

      return true;
    } catch (error) {
      print('error creating filter: $error');
      return false;
    }
  }

  Future<bool> delete(String filterId) async {
    try {
      await dio.api.delete('/filters/$filterId');
      return true;
    } catch (error) {
      print('error deleting filter: $error');
      return false;
    }
  }
}
